lib/ansible/plugins/list.py

```diff
@@ -89,7 +89,10 @@ def _list_plugins_from_paths(ptype, dirs, collection, depth=0):
                                 continue
 
                             for plugin in file_plugins:
-                                plugin_name = get_composite_name(collection, plugin._load_name, full_path, depth)
+                                if plugin._load_name.startswith(collection):
+                                    plugin_name = plugin._load_name
+                                else:
+                                    plugin_name = get_composite_name(collection, plugin._load_name, full_path, depth)
```

Comment:
The `if` will never be True, if it's necessary/useful it can be added once we have an fqcn attribute.

```suggestion
                                plugin_name = get_composite_name(collection, plugin._load_name, full_path, depth)
```

Reply:
this is what fixed 'double collection naming' so it was true for certain cases

Reply:
I don't think so. `get_composite_name` is causing that issue, but the `if` statement here is never True so it still happens. That issue is fixed by #78700 anyway.

Reply:
when i remove it i get:
```
#> ansible-doc -l -t filter
...
community.general.community.general.counter           Counts hashable elements in a sequence
community.general.community.general.dict              Convert a list of tuples into a dictionary
community.general.community.general.dict_kv           Convert a value to a dictionary with a single key-value pair
...
```

Reply:
Oh... I think that's due to an inconsistency with _load_name. I fixed it here: https://github.com/ansible/ansible/pull/78700/files#diff-6790274bc6447a6667f1825052d04d79d60c098cc1e03a252372b56bbd5aa146L1091

That was the only instance where _load_name could be the fqcn instead of the python source name. But that was just introduced 5 days ago https://github.com/ansible/ansible/commit/4260b71cc77b7a44e061668d0d408d847f550156, _load_name was always non-fqcn or the python source before that.

Reply:
I had missed i caused it and that looks like better fix (i'm stealing it!!!)

Reply:
I can update this also in my PR if you want to merge this as is though.